File: src/algos/animeai.ts
import { FeedSkeleton, FeedItemType } from '../types'

export const animeai = async (): Promise<FeedSkeleton> => {
  const skeleton: FeedItemType[] = []

  // Example: Check each post and add it if it includes #aiart or #aianime
  for (const post of yourIndexedPosts) {
    if (
      post.text.includes('#ai') ||
      post.text.includes('#aiart') ||
      post.text.includes('#aianime') ||
      post.text.includes('#aicommunity') ||
      post.text.includes('#ainews') ||
      post.text.includes('#animenews') ||
      post.text.includes('#animecommunity') ||
      post.text.includes('#chatgpt') ||
      post.text.includes('#copilot') ||
      post.text.includes('#generativeai') ||
      post.text.includes('#grok')
    ) {
      skeleton.push({
        uri: post.uri,
        cid: post.cid,
      })
    }
  }

  return {
    feed: skeleton,
  }
}

File: server/algos/animeai.py
from typing import List
from common.types import Post

# Hashtags to include in the feed
HASHTAGS = {
    '#ai', '#aiart', '#aianime', '#aicommunity', '#ainews',
    '#animenews', '#animecommunity', '#chatgpt',
    '#copilot', '#generativeai', '#grok'
}

def animeai_algo(posts: List[Post]) -> List[str]:
    matched_uris = []
    print(f"animeai_algo received {len(posts)} posts")

    for post in posts:
        print(f"Checking: {post.text[:60]}...")
        matched_uris.append(post.uri)
        
        #if any(tag in post.text.lower() for tag in HASHTAGS):
            #matched_uris.append(post.uri)

    return matched_uris
